File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.java
class Solution {
    public String solution(String X, String Y) {
        int[] cx = new int[10];
        int[] cy = new int[10];
        
        for (char c : X.toCharArray()) cx[c - '0']++;
        for (char c : Y.toCharArray()) cy[c - '0']++;
        
        StringBuilder sb = new StringBuilder();
        for (int n = 9; n >= 0; n--) {
            int target = Math.min(cx[n], cy[n]);
            while (target-- > 0) sb.append(n);
        }
        
        if (sb.length() == 0) return "-1";
        if (sb.charAt(0) == '0') return "0";
        
        return sb.toString();
    }
}

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.kt
class Solution {
    fun solution(X: String, Y: String): String {
        val cx = IntArray(10)
        val cy = IntArray(10)
        
        for (c in X) cx[c - '0']++
        for (c in Y) cy[c - '0']++
        
        val sb = StringBuilder()
        for (n in 9 downTo 0) {
            val cnt = minOf(cx[n], cy[n])
            repeat(cnt) { sb.append(n) }
        }
        
        if (sb.isEmpty()) return "-1"
        if (sb[0] == '0') return "0"
        
        return sb.toString()
    }
}